### PR TITLE
NIFI-8656: Support expression language for SAS Token in the ADLS Gen2…

### PR DESCRIPTION
**Pull request by turcsanyip**
… processors

https://issues.apache.org/jira/browse/NIFI-8656

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [x] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by jfrazee** [comment]
@turcsanyip On paper this LGTM. I'll try to give it a test.

This ties into the discussion in #4843 I think, right? So we're walking back the change just for SAS tokens. I think that makes sense and is a good compromise because the use case is a solid one and SAS tokens present much less risk than the access key.

Would it make sense to add more detail to `additionalDetails.html` on the trade offs and maybe suggestions around paying attention to queue and provenance policies?

File: nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/services/azure/storage/TestADLSCredentialsControllerService.java
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/services/azure/storage/TestADLSCredentialsControllerService.java:341]: Although a few tests include the given/when/then comments, I recommend avoiding them for new tests.  If there are no other changes, then it is fine to leave them in place, but otherwise I think it would be better to avoid the comments.

**Review by exceptionfactory — COMMENTED** [review]
Looks like a straightforward change. Although supporting FlowFile attributes for these types of properties can introduce security concerns, the short lifespan of the SAS Token and the property description seem sufficient to allow the option for this particular property.

File: nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/services/azure/storage/TestADLSCredentialsControllerService.java
Comment (stan-antyufeev-8451) [reply nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/services/azure/storage/TestADLSCredentialsControllerService.java:341]: Hi 
Can we merge this change to proceed further?  I need this feature to test our on one of the solution built on Nifi.

**Comment by jfrazee** [comment]
@stan-antyufeev-8451 Are you able to or have you given if a test? I'm happy to merge it if so. I can't imagine any issue given the change, but I try to not merge without trying stuff out and hadn't had a chance to.

**Comment by turcsanyip** [comment]
@jfrazee Added Additional Details page to explain the security considerations. Feel free to suggest more items that could be worth mentioning.
Also changed the descriptions of some properties because the security concerns regarding EL were displayed for some properties that do not support EL at all.

@exceptionfactory Removed the comments from the test.

**Comment by stan-antyufeev-8451** [comment]
@jfrazee  I tested locally and it works for me. I need to promote this fix to our Nifi cluster. Appreciated you work, guys.

**Comment by stan-antyufeev-8451** [comment]
@jfrazee  Anything  else do you have or You can merge to main branch ?

**Comment by jfrazee** [comment]
@exceptionfactory I'm going to do another pass through this today. Did you have any additional concerns or feedback on this?

**Comment by exceptionfactory** [comment]
> @exceptionfactory I'm going to do another pass through this today. Did you have any additional concerns or feedback on this?

Thanks for asking @jfrazee, looks good from my perspective, feel free to merge if it looks good to you.

**Review by exceptionfactory — APPROVED** [review]

**Review by jfrazee — APPROVED** [review]

**Pull request closed without merging**
